app/controllers/tickets_controller.rb

```diff
@@ -9,22 +9,13 @@ def show
       end
       format.json do
         ticket = Ticket.find(params.require(:id))
-        # requester_stakeholders will have the list of stakeholders where the requester is part of.
-        # For example, if a normal user X requests for a change by creating a ticket, the
-        # stakeholders list will be [X, WRT] (WRT is added as stakeholder because WRT is
-        # responsible for taking action on the ticket). If a WRT member fetches the ticket data,
-        # the value of requester_stakeholders will be [WRT] and if the normal user fetches the
-        # ticket data, the value of requester_stakeholders will be [X]. If the ticket is created by
-        # a WRT member, then the value requester_stakeholders will be [X, WRT] because the user can
-        # be any of the two stakeholders.
-        requester_stakeholders = ticket.user_stakeholders(current_user)
 
         # Currently only stakeholders can access the ticket.
-        return head :unauthorized if requester_stakeholders.empty?
+        return head :unauthorized if ticket.can_user_access?(current_user)
```

Comment:
Ehrm... I was about to merge this until I realized: Shouldn't this be an `unless`?!

Reply:
Oops sorry yes. This was a big miss, sorry. I've fixed it now.